src/client/rpc/tonic_client.rs

```diff
@@ -230,14 +221,32 @@ impl NodeRpcClient for TonicRpcClient {
             "GetAccountDetails response should have an `account`".to_string(),
         ))?;
 
-        let details_bytes =
-            account_info.details.ok_or(NodeRpcClientError::ExpectedFieldMissing(
-                "GetAccountDetails response's account should have `details`".to_string(),
+        let account_summary =
+            account_info.summary.ok_or(NodeRpcClientError::ExpectedFieldMissing(
+                "GetAccountDetails response's account should have a `summary`".to_string(),
             ))?;
 
-        let details = Account::read_from_bytes(&details_bytes)?;
+        let hash = account_summary.account_hash.ok_or(NodeRpcClientError::ExpectedFieldMissing(
+            "GetAccountDetails response's account should have an `account_hash`".to_string(),
+        ))?;
+
+        let hash = Digest::try_from([hash.d0, hash.d1, hash.d2, hash.d3]).map_err(|_| {
+            NodeRpcClientError::InvalidAccountReceived("Invalid account hash".into())
+        })?;
```

Comment:
You can use the conversion between digest defined in miden_node_proto
```suggestion
        let hash = hash.try_into()?;
```